dataprofiler/profilers/profile_builder.py

```diff
@@ -1439,54 +1485,108 @@ def tqdm(l):
 
         if pool is not None:
             # Create a bunch of simultaneous column conversions
-            for col in data.columns:
-                if min_true_samples is None:
-                    min_true_samples = self._profile[col]._min_true_samples
-                try:
-                    multi_process_dict[col] = pool.apply_async(
-                        self._profile[col].clean_data_and_get_base_stats,
-                        (data[col], sample_size, min_true_samples, sample_ids))
-                except Exception as e:
-                    print(e)
-                    single_process_list.add(col)
+            for col in data.columns.unique():
```

Comment:
instead of looping by column name, just loop through by column index in the df?

Reply:
I think the if statements maybe able to removed if this is the case and / or the code can be condensed.